finagle-core/src/main/scala/com/twitter/finagle/util/InetSocketAddressUtil.scala

```diff
@@ -1,14 +1,49 @@
 package com.twitter.finagle.util
 
-import java.net.{InetAddress, InetSocketAddress, SocketAddress, UnknownHostException}
+import java.net.{Inet4Address, Inet6Address, InetAddress, InetSocketAddress,
+  NetworkInterface, SocketAddress, SocketException, UnknownHostException}
+import com.twitter.logging.Logger
 
 object InetSocketAddressUtil {
 
   type HostPort = (String, Int)
 
+  private[this] val log = Logger()
+
   private[finagle] val unconnected =
     new SocketAddress { override def toString = "unconnected" }
 
+  private[this] lazy val anyInterfaceSupportsIpV6: Boolean = {
```

Comment:
is there a reason this has to be lazy?

Reply:
We fixed an internal bug at some point by putting lazy to. a field of a companion object. 
(AFAIR there was a cyclic dependency).
Once fixed, a senior member of the team told us, that we'd rather put lazy by defaults in this case. 
Still given it's a local policy and that I did not find any external reference mentioning that practice, I should not have used it here. Sorry for that, I removed the lazy :)